ras/ras_ppcutils.py

```diff
@@ -447,3 +449,40 @@ def test_lparnumascore(self):
         for list_item in lists:
             self.run_cmd('lparnumascore %s' % list_item)
         self.error_check()
+
+    @skipIf(IS_POWER_NV or IS_KVM_GUEST,
+            "This test is not supported on KVM guest or PowerNV platform")
+    def test_validate_lparstat(self):
+        """
+        Test case validates laprstat -E output %busy and %idle
+        should not be < 0 or > 100
+        Normalized %busy + %idle should be equal to percentage under frequency
+        """
+        output = process.system_output("lparstat -E 1 1").decode("utf-8")
+        input_string = ''
+        for line in output.splitlines():
+            if 'GHz' in line:
+                input_string = line
+                # Define the regular expression pattern
+                pattern = r'(\d+\.\d+)\s+(\d+\.\d+)\s+\d+\.\d+GHz\[\s*(\d+)%\]\s+(\d+\.\d+)\s+(\d+\.\d+)'
+                # Find all matches in the input string
+                matches = re.findall(pattern, input_string)
+                for data in matches:
+                    actual_busy = float(data[0])
+                    actual_idle = float(data[1])
+                    normal_idle = float(data[3])
+                    normal_busy = float(data[4])
+                    normal = normal_idle + normal_busy
+                    freq_percentile = float(data[2])
+        if (actual_busy > 0) and (actual_idle < 100):
+            self.log.info(
+                "Actual busy and idle values are displaying  correctly")
+        else:
+            self.cancel(
+                "busy value and idle values are not  displaying correctly")
+        if normal == freq_percentile:
+            self.log.info("normal busy and idle are matching with the \
+                           percentage under frequency")
+        else:
+            self.cancel("normal busy and value percentage are not matching \
+                        with the frequency percentile")
```

Comment:
here too should fail instead of cancel ?

Reply:
yeah, for the above two cases I have changed it to fail if the condition is not met